src/main/scala/is/hail/variant/Genotype.scala

```diff
@@ -90,97 +73,100 @@ class Genotype(private val _gt: Int,
     fakeRef: Boolean = this.fakeRef,
     isDosage: Boolean = this.isDosage): Genotype = Genotype(gt, ad, dp, gq, px, fakeRef, isDosage)
 
-  override def equals(that: Any): Boolean = that match {
+
+  override def equals(that: Any): Boolean =
+    that match {
+
     case g: Genotype =>
-      _gt == g._gt &&
-        ((_ad == null && g._ad == null)
-          || (_ad != null && g._ad != null && _ad.sameElements(g._ad))) &&
-        _dp == g._dp &&
-        _gq == g._gq &&
+      unboxedGT == g.unboxedGT &&
+        ((ad.isEmpty && g.ad.isEmpty) || (ad.isDefined && g.ad.isDefined && ad.get.sameElements(g.ad.get))) &&
+        dp == g.dp &&
+        gq == g.gq &&
+        ((px.isEmpty && g.px.isEmpty) || (px.isDefined && g.px.isDefined && px.get.sameElements(g.px.get))) &&
         fakeRef == g.fakeRef &&
-        isDosage == g.isDosage &&
-        ((_px == null && g._px == null)
-          || (_px != null && g._px != null && _px.sameElements(g._px)))
+        isDosage == g.isDosage
 
     case _ => false
   }
 
   override def hashCode: Int =
     new HashCodeBuilder(43, 19)
-      .append(_gt)
-      .append(util.Arrays.hashCode(_ad))
-      .append(_dp)
-      .append(_gq)
-      .append(util.Arrays.hashCode(_px))
+      .append(unboxedGT)
+      .append(if (ad.isEmpty) null else util.Arrays.hashCode(ad.get))
```

Comment:
This probably isn't what you want.  hashCode returns an int, null is not.  You probably want `util.Arrays.hashCode(ad.getOrElse(null))`.  `hashCode(int a[])` handles null appropriately (hashes to 0).